plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -935,9 +940,72 @@ public Optional<ConnectorTableHandle> applyDelete(ConnectorSession session, Conn
     }
 
     @Override
-    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle, RetryMode retryMode)
```

Comment:
@losipiuk do I need to do something with this `RetryMode` argument?

Reply:
I would expect you need to pass it via tableHandle to `finishDelete` so you know that you should do a cleanup of potentially leftover files there.